python/pyspark/sql/dataframe.py

```diff
@@ -5797,25 +5798,53 @@ def withColumnRenamed(self, existing: str, new: str) -> "DataFrame":
         Parameters
         ----------
         existing : str
-            string, name of the existing column to rename.
+            The name of the existing column to be renamed.
         new : str
-            string, new name of the column.
+            The new name to be assigned to the column.
 
         Returns
         -------
         :class:`DataFrame`
-            DataFrame with renamed column.
+            A new DataFrame with renamed column.
+
+        See Also
+        --------
+        :meth:`withColumnsRenamed`
 
         Examples
         --------
         >>> df = spark.createDataFrame([(2, "Alice"), (5, "Bob")], schema=["age", "name"])
-        >>> df.withColumnRenamed('age', 'age2').show()
+
+        Example 1: Rename a single column
+
+        >>> df.withColumnRenamed("age", "age2").show()
         +----+-----+
         |age2| name|
         +----+-----+
         |   2|Alice|
         |   5|  Bob|
         +----+-----+
+
+        Example 2: Rename a column that does not exist (no-op)
+
+        >>> df.withColumnRenamed("non_existing", "new_name").show()
+        +---+-----+
+        |age| name|
+        +---+-----+
+        |  2|Alice|
+        |  5|  Bob|
+        +---+-----+
+
+        Example 3: Rename multiple columns
+
+        >>> df.withColumnRenamed("age", "age2") \\
+        ...     .withColumnRenamed("name", "name2").show()
```

Comment:
actually, let's don't use `\\`. That's sort of discouraged by PEP 8., and seems it fits within 100 characters.

```suggestion
        >>> df.withColumnRenamed("age", "age2").withColumnRenamed("name", "name2").show()
```